### PR TITLE
Ported 'a's to python

### PR DESCRIPTION
**Pull request by Twinters007**

### CODE REVIEW

File: wpilib/wpilib/analoggyro.py [hunk 1]
```diff
@@ -84,81 +79,57 @@ def __init__(self, channel, center=None, offset=None):
         if center is None or offset is None:
             self.calibrate()
         else:
-            self.center = int(center)
-            self.offset = float(offset)
-            self.analog.setAccumulatorCenter(self.center)
-            self.analog.resetAccumulator()
+            hal.setAnalogGyroParameters(self._gyroHandle, self.kDefaultVoltsPerDegreePerSecond, offset, center)        
+            reset()
```
Comment (virtuald) [inline wpilib/wpilib/analoggyro.py]: `self.reset`?

File: wpilib/wpilib/analoggyro.py [hunk 2]
```diff
@@ -64,18 +64,13 @@ def __init__(self, channel, center=None, offset=None):
...
-        self.voltsPerDegreePerSecond = AnalogGyro.kDefaultVoltsPerDegreePerSecond
-        self.analog.setAverageBits(AnalogGyro.kAverageBits)
-        self.analog.setOversampleBits(AnalogGyro.kOversampleBits)
-        sampleRate = AnalogGyro.kSamplesPerSecond \
-                * (1 << (AnalogGyro.kAverageBits + AnalogGyro.kOversampleBits))
-        AnalogInput.setGlobalSampleRate(sampleRate)
-        if not hal.HALIsSimulation():
-            Timer.delay(1.0)
-        
+        if self._gyroHandle == 0:
```
Comment (virtuald) [inline wpilib/wpilib/analoggyro.py]: I don't think this gets initialized before this line?

File: wpilib/wpilib/analoggyro.py [hunk 3]
```diff
@@ -84,81 +79,57 @@ def __init__(self, channel, center=None, offset=None):
...
+        hal.resetAnalogGyro(self._gyroHandle)
 
     def free(self):
         """:see: :meth:`.Gyro.free`"""
         LiveWindow.removeComponent(self)
         if self.analog is not None and self.channelAllocated:
             self.analog.free()
             self.analog = None
-
+        hal.freeAnalogGyro(self._gyroHandle)
```
Comment (virtuald) [inline wpilib/wpilib/analoggyro.py]: Should set handle to None after this

File: wpilib/wpilib/analoggyro.py [hunk 4]
```diff
@@ -64,18 +64,13 @@ def __init__(self, channel, center=None, offset=None):
...
-        self.analog.setAverageBits(AnalogGyro.kAverageBits)
-        self.analog.setOversampleBits(AnalogGyro.kOversampleBits)
-        sampleRate = AnalogGyro.kSamplesPerSecond \
-                * (1 << (AnalogGyro.kAverageBits + AnalogGyro.kOversampleBits))
-        AnalogInput.setGlobalSampleRate(sampleRate)
-        if not hal.HALIsSimulation():
-            Timer.delay(1.0)
-        
+        if self._gyroHandle == 0:
+            self._gyroHandle = hal.initializeAnalogGyro(self.analog.port)
```
Comment (virtuald) [inline wpilib/wpilib/analoggyro.py]: By convention, we usually turn `m_foo` into `self.foo`

File: wpilib/wpilib/analogoutput.py [hunk 5]
```diff
@@ -28,13 +28,8 @@ def __init__(self, channel):
...
         :param channel: The channel number to represent.
         """
-        if not hal.checkAnalogOutputChannel(channel):
-            raise IndexError("Analog output channel %d cannot be allocated. Channel is not present." % channel)
-        try:
-            AnalogOutput.channels.allocate(self, channel)
-        except IndexError as e:
-            raise IndexError("Analog output channel %d is already allocated" % channel) from e
-
+        SensorBase.checkANalogOutputChannel(channel)
```
Comment (virtuald) [inline wpilib/wpilib/analogoutput.py]: Typo.

File: wpilib/wpilib/analogpotentiometer.py [hunk 6]
```diff
@@ -49,7 +49,7 @@ def __init__(self, channel, fullRange=1.0, offset=0.0):
         if not hasattr(channel, "getVoltage"):
             channel = AnalogInput(channel)
         self.analog_input = channel
-        self.fullRange = fullRange
+        self._fullRange = fullRange
```
Comment (virtuald) [inline wpilib/wpilib/analogpotentiometer.py]: Why did this get changed?

File: wpilib/wpilib/analoggyro.py (hunk 4 above)
Comment (Twinters007) [inline wpilib/wpilib/analoggyro.py]: it was a private variable so I added the _ just in case. Should I change it?

File: wpilib/wpilib/analogpotentiometer.py (hunk 6 above)
Comment (Twinters007) [inline wpilib/wpilib/analogpotentiometer.py]: became private

File: wpilib/wpilib/analoggyro.py (hunk 4 above)
Comment (virtuald) [inline wpilib/wpilib/analoggyro.py]: I've thought about using that convention, but I think the rest of the code doesn't do that, so let's not start now.

Comment (Twinters007) [inline wpilib/wpilib/analoggyro.py]: lol

File: wpilib/wpilib/analoggyro.py (hunk 3 above)
Comment (Twinters007) [inline wpilib/wpilib/analoggyro.py]: since the code above checks for 0 i'll set to 0

File: wpilib/wpilib/analogpotentiometer.py [hunk 7]
```diff
@@ -60,7 +60,7 @@ def get(self):
         :returns: The current position of the potentiometer.
         :rtype: float
         """
-        return (self.analog_input.getVoltage() / hal.getUserVoltage5V()) * self.fullRange + self.offset
+        return (self.analog_input.getVoltage() / hal.getUserVoltage5V()) * self._fullRange + self._offset
```
Comment (virtuald) [inline wpilib/wpilib/analogpotentiometer.py]: Forgot one.

File: wpilib/wpilib/analoggyro.py [hunk 8]
```diff
@@ -37,7 +37,7 @@ class AnalogGyro(GyroBase):
     kDefaultVoltsPerDegreePerSecond = 0.007
     
     PIDSourceType = PIDSource.PIDSourceType
-
+    gyroHandle = 0
```
Comment (virtuald) [inline wpilib/wpilib/analoggyro.py]: Spacing, should have at least one line above each function.

Also... this feels a bit odd to me. It's not *wrong*. But. I'd rather see this assigned in the constructor.

File: wpilib/wpilib/analoggyro.py [hunk 9]
```diff
@@ -64,101 +67,73 @@ def __init__(self, channel, center=None, offset=None):
...
-            self.center = int(center)
-            self.offset = float(offset)
-            self.analog.setAccumulatorCenter(self.center)
-            self.analog.resetAccumulator()
+            hal.setAnalogGyroParameters(self.gyroHandle, self.kDefaultVoltsPerDegreePerSecond, offset, center)        
+            self.reset()
             
-
+         
+        self.gyroHandle = 0
```
Comment (virtuald) [inline wpilib/wpilib/analoggyro.py]: Why is this here?

File: wpilib/wpilib/analoggyro.py [hunk 10]
```diff
@@ -64,101 +67,73 @@ def __init__(self, channel, center=None, offset=None):
...
 
     def free(self):
         """:see: :meth:`.Gyro.free`"""
         LiveWindow.removeComponent(self)
         if self.analog is not None and self.channelAllocated:
             self.analog.free()
             self.analog = None
-
+        hal.freeAnalogGyro(self.gyroHandle)
+        self.gyroHandle = 0
```
Comment (virtuald) [inline wpilib/wpilib/analoggyro.py]: Since there are changes to be made... spacing!

File: wpilib/wpilib/analogpotentiometer.py [hunk 11]
```diff
@@ -60,7 +60,7 @@ def get(self):
         :returns: The current position of the potentiometer.
         :rtype: float
         """
-        return (self.analog_input.getVoltage() / hal.getUserVoltage5V()) * self.fullRange + self.offset
+        return (self.analog_input.getVoltage() / hal.getUserVoltage5V()) * self.fullRange + self._offset
```
Comment (virtuald) [inline wpilib/wpilib/analogpotentiometer.py]: Somehow I missed this (perhaps because of fullrange), but line 53 says that offset doesn't start with an underscore.

File: wpilib/wpilib/analoggyro.py (hunk 9 above)
Comment (Twinters007) [inline wpilib/wpilib/analoggyro.py]: couldn't tell you
